### PR TITLE
fix: fix app crashing while calculating price impact

### LINKED ISSUE #2467: Crash report: `TypeError: Invalid attempt to destructure non-iterable instance.In order to be iterable, non-array objects must have a [Symbol.iterator]() method.`
## URL

https://swap.cow.fi/#/1/limit-orders/USDC/USDT?tab=open&page=1

## Error

```
TypeError: Invalid attempt to destructure non-iterable instance.
In order to be iterable, non-array objects must have a [Symbol.iterator]() method.
```

## Stacktrace

```
TypeError: Invalid attempt to destructure non-iterable instance.
In order to be iterable, non-array objects must have a [Symbol.iterator]() method.
    at r (https://swap.cow.fi/static/js/main.a7999048.js:2:3877412)
    at a (https://swap.cow.fi/static/js/main.a7999048.js:2:3886502)
    at A (https://swap.cow.fi/static/js/main.a7999048.js:2:1063435)
    at https://swap.cow.fi/static/js/main.a7999048.js:2:884742
    at ru (https://swap.cow.fi/static/js/main.a7999048.js:2:3351912)
    at Ac (https://swap.cow.fi/static/js/main.a7999048.js:2:3372139)
    at uc (https://swap.cow.fi/static/js/main.a7999048.js:2:3364396)
    at zi (https://swap.cow.fi/static/js/main.a7999048.js:2:3305095)
    at https://swap.cow.fi/static/js/main.a7999048.js:2:3370935
    at xc (https://swap.cow.fi/static/js/main.a7999048.js:2:3370940)
    at ic (https://swap.cow.fi/static/js/main.a7999048.js:2:3363398)
    at A (https://swap.cow.fi/static/js/main.a7999048.js:2:3588322)
    at MessagePort.P (https://swap.cow.
```

## Device data

```json
{
  "ua": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/113.0.0.0 Safari/537.36",
  "browser": {
    "name": "Chrome",
    "version": "113.0.0.0",
    "major": "113"
  },
  "engine": {
    "name": "Blink",
    "version": "113.0.0.0"
  },
  "os": {
    "name": "Windows",
    "version": "10"
  },
  "device": {},
  "cpu": {
    "architecture": "amd64"
  }
}
```

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Fixes #2467, #2510

image

The function `calculateFallbackPriceImpact` throws an error when `initialValue` equals `"0"` (because of the division on zero).
Since the function is called only from `useFallbackPriceImpact` hook, the check for `!== "0"` is added here.

# Background
I was able to reproduce it only when applied the issue reporters `localStorage` snapshot.
My assumption is:
 - the reporter has cached some `wrong values` in his `localStorage`
 - those `wrong values` influence on price impact calculations and cause calling `useFallbackPriceImpact` (normally it shouldn't be called, since it's a fallback)
 - `calculateFallbackPriceImpact` didn't consider a case when `initialValue` is zero as string
 
 UPD:
 Elena found one more issue related to `"0"` value: https://github.com/cowprotocol/cowswap/pull/2505#issuecomment-1557009718. `calculateExecutionPrice` also fails when `inputCurrencyAmount` is zero

### CODE REVIEW

**Comment by elena-zh** [comment]
hey @shoom3301 , the changes LGTM when I open the app in Ethereum chain. 
However, I can receive the app crash when I open the app in Goerli, and select USDC-USDT token pair. 
image

**Review by nenadV91 — APPROVED** [review]

**Comment by nenadV91** [comment]
Soft approve for code, please address Elena comment

**Comment by shoom3301** [comment]
@elena-zh thanks! Fixed

**Review by anxolin — APPROVED** [review]
Approved! just wait for Elena's check before merging :)

**Comment by elena-zh** [comment]
One more crash: being connected to EOA open Limit orders page, connect to Goerli.  
Select USDC-USDT token pair, an amount ot trade
Open the token list. switch a network to an unsupported one (I switched to Volta)
image

**Review by elena-zh — APPROVED** [review]
Looks like all crashes are no longer reproducible here!

**Pull request merged**
